safe-global/safe-wallet-web
# feat: Add Human readable descriptions

## What it solves

First part of the EPIC #2011 

Resolves #2448 

Depends on https://github.com/safe-global/safe-client-gateway-nest/pull/631

## How this PR fixes it

- Displays human-readable descriptions for nonce replacement and notifications

## ToDos

- [x] Update gateway-typescript-sdk type for `TransactionInfo`
- [x] Check remaining `dispatch` functions if they need a `txDescription`


## How to test it

1. Queue a transaction (transfer, settings change, custom transaction)
2. Create a new transaction
3. Open the nonce replacement form
4. Observe human-readable descriptions
5. If there is no human-readable description observe the same message as before
6. Sign or execute a transaction
7. Observe a human-readable description inside notifications and the notification center
8. If there is no human-readable description observe a generic fallback saying `Transaction` as the Notification title

## Screenshots
<img width="366" alt="Screenshot 2023-08-11 at 15 41 52" src="https://github.com/safe-global/safe-wallet-web/assets/5880855/ea503859-61b4-434d-9083-74976c82fca6">
<img width="471" alt="Screenshot 2023-08-11 at 17 22 29" src="https://github.com/safe-global/safe-wallet-web/assets/5880855/4a0510d2-df41-40dc-a8a2-e50a4716bff8">

<img width="380" alt="Screenshot 2023-08-11 at 13 18 53" src="https://github.com/safe-global/safe-wallet-web/assets/5880855/cf50ab45-b31f-4688-a459-eb98b59b363d">


## Checklist
